Add DualStack tests to psm_interop_kokoro_lib.sh

aranjans opened the pull request:
Added dualstack test to  `psm_interop_kokoro_lib.sh`.

purnesh42H commented on .kokoro/psm_interop_kokoro_lib.sh:
```diff
@@ -694,6 +739,10 @@ activate_gke_cluster() {
       ;;
+    GKE_CLUSTER_DUAL_STACK)
+          GKE_CLUSTER_NAME="psm-interop-lb-primary-ds"
+          GKE_CLUSTER_ZONE="us-central1-c"
```
zone should us-central1-a

sergiitk commented on .kokoro/psm_interop_kokoro_lib.sh:
```diff
@@ -694,6 +739,10 @@ activate_gke_cluster() {
       GKE_CLUSTER_NAME="interop-test-psm-basic"
       GKE_CLUSTER_ZONE="us-central1-c"
       ;;
+    GKE_CLUSTER_DUAL_STACK)
+          GKE_CLUSTER_NAME="psm-interop-lb-primary-ds"
+          GKE_CLUSTER_ZONE="us-central1-a"
+          ;;
```
```suggestion
    GKE_CLUSTER_DUAL_STACK)
      GKE_CLUSTER_NAME="psm-interop-lb-primary-ds"
      GKE_CLUSTER_ZONE="us-central1-a"
      ;;
```

The indentation is off - please fix

sergiitk commented on .kokoro/psm_interop_kokoro_lib.sh:
```diff
@@ -694,6 +739,10 @@ activate_gke_cluster() {
       GKE_CLUSTER_ZONE="us-central1-c"
       ;;
+    GKE_CLUSTER_DUAL_STACK)
+          GKE_CLUSTER_NAME="psm-interop-lb-primary-ds"
```
Would be great if we used the new cluster name here
```suggestion
          GKE_CLUSTER_NAME="psm-interop-dualstack"
```

sergiitk commented on .kokoro/psm_interop_kokoro_lib.sh:
```diff
@@ -694,6 +739,10 @@ activate_gke_cluster() {
       GKE_CLUSTER_NAME="interop-test-psm-basic"
       GKE_CLUSTER_ZONE="us-central1-c"
       ;;
+    GKE_CLUSTER_DUAL_STACK)
```
nit
```suggestion
    GKE_CLUSTER_DUALSTACK)
```

sergiitk commented on .kokoro/psm_interop_kokoro_lib.sh:
```diff
@@ -158,6 +158,51 @@ psm::security::run_test() {
+#######################################
+psm::dual_stack::get_tests() {
+  TESTS=(
+    "test_dual_stack"
```
```suggestion
    "test_dualstack"
```

sergiitk commented on .kokoro/psm_interop_kokoro_lib.sh:
```diff
@@ -158,6 +158,51 @@ psm::security::run_test() {
+#######################################
+psm::dual_stack::run_test() {
+  local test_name="${1:?${FUNCNAME[0]} missing the test name argument}"
+
+  # Only java supports extra checks for certificate matches (via channelz socket info).
+  if [[ "${GRPC_LANGUAGE}" != "java"  ]]; then
+    PSM_TEST_FLAGS+=("--nocheck_local_certs")
+  fi
```
This is only needed for security tests

```suggestion
```

sergiitk commented on .kokoro/psm_interop_kokoro_lib.sh:
```diff
@@ -158,6 +158,51 @@ psm::security::run_test() {
+  if [[ "${GRPC_LANGUAGE}" != "java"  ]]; then
+    PSM_TEST_FLAGS+=("--nocheck_local_certs")
+  fi
+
```
```suggestion
  PSM_TEST_FLAGS+=(
    "--flagfile=config/common-dualstack.cfg"
  )
```

Let's create config/common-dualstack.cfg, with the following content:

```
# Common config file for PSM DualStack tests.
--resource_prefix=psm-ds
--enable_dualstack
--noenable_workload_identity
```

sergiitk commented on .kokoro/psm_interop_kokoro_lib.sh:
```diff
@@ -158,6 +158,51 @@ psm::security::run_test() {
+# Outputs:
+#   Prints activated cluster names.
+#######################################
+psm::dual_stack::setup() {
```
Here and elsewhere: rename `dual_stack` to `dualstack`.
This is to be consistent with how we already refer to this functionality in the framework (e.g. `--enable_dualstack` flag).

aranjans commented:
@sergiitk thAnks for your review, I have updated the PR with all the suggestions.
fyi: I have updated the GKE cluster name but @purnesh42H is yet to create the new cluster with same name.
